dts/bindings/gpio/nordic,nrf-gpio-forwarder.yaml

```diff
@@ -0,0 +1,59 @@
+# Copyright (c) 2020 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    This is an abstract device responsible for forwarding pins between cores.
+
+    Here is an example of how a nrf-gpio-forwarder can be used with a nRF5340
+    combined with a nRF21540 Front-End module. Consider the following node
+    present in DTS file targeted for the nRF5340 network core:
+
+    nrf_radio_fem: nrf21540 {
+      compatible = "nordic,nrf21540-fem";
+      tx-en-gpios = <&gpio0 30 GPIO_ACTIVE_HIGH>;
+      rx-en-gpios = <&gpio1 11 GPIO_ACTIVE_HIGH>;
+      pdn-gpios   = <&gpio1 10 GPIO_ACTIVE_HIGH>;
+      mode-gpios  = <&gpio1 12 GPIO_ACTIVE_HIGH>;
+    };
+
+    and the following node defined in DTS file targeted for the nRF5340
+    application core:
+
+    gpio_fwd: nrf-gpio-forwarder {
+      compatible = "nordic,nrf-gpio-forwarder";
+      status = "okay";
+      nrf21540-gpio-if {
+        gpios = <&gpio0 30 0>, <&gpio1 11 0>, <&gpio1 10 0>, <&gpio1 12 0>;
+      };
+    };
+
+    In the above example, the nrf-gpio-forwarder node is configured to forward
+    control over the following GPIO pins:
+
+      - P0.30 (from `tx-en-gpios`)
+      - P1.11 (from `rx-en-gpios`)
+      - P1.10 (from `pdn-gpios`)
+      - P1.12 (from `mode-gpios`)
+
+    Please note that the GPIO flags provided for child nodes of the forwarder
+    are ignored. In order to configure the GPIOs passed to the forwarder, their
+    GPIO flags must be set in the matching node that these GPIOs are forwarded
+    to. In the above example, the GPIO flags must be set in the nrf21540 node.
+    They are set to 0 in the nrf-gpio-forwarder node as they are ignored anyway.
```

Comment:
You provide quite detailed doc. Could you please provide additional info on how to forward pins if there are more net-core nodes as `nrf_radio_fem` having gpio to be forwarded from app core, and these nodes do not know of each other?  
Are there any requirements on naming nodes within `nrf-gpio-forwarder` ?

Reply:
Ok, so please document this fact.

Reply:
@ankuns I added a section to the binding doc.